src/annotator/highlighter.ts

```diff
@@ -306,6 +307,38 @@ export function removeHighlights(highlights: HighlightElement[]) {
   }
 }
 
+/**
+ * Focus or un-focus an individual SVG highlight element.
+ *
+ * When focusing an SVG highlight, make sure it is not obscured by other SVG
+ * highlight elements. As SVG highlights are siblings, this can be accomplished
+ * by putting the highlight at the end the set of highlights contained its
+ * parent. SVG highlight elements are cloned instead of moved so that their
+ * original stacking (nesting) order is not lost when later unfocused. A data
+ * attribute is added to associate the original SVG highlight element with its
+ * clone.
+ */
+function setSVGHighlightFocused(svgEl: SVGElement, focused: boolean) {
+  const parent = svgEl.parentNode as SVGElement;
+  const focusedId = svgEl.getAttribute('data-focused-id');
+  if (focused) {
+    if (!focusedId) {
+      svgEl.setAttribute('data-focused-id', generateHexString(8));
+    }
+    const focusedHighlight = svgEl.cloneNode() as SVGRectElement;
+    focusedHighlight.classList.add('is-focused');
+    parent.append(focusedHighlight);
+  } else if (focusedId) {
+    const focusedHighlight = parent.querySelector(
+      `[data-focused-id="${focusedId}"].is-focused`
+    );
+    if (focusedHighlight) {
+      focusedHighlight.remove();
+    }
+    svgEl.removeAttribute('data-focused-id');
```

Comment:
The temporary focused element gets removed when the highlight is un-focused in the sidebar, but not if the highlight is removed for another reason with `removeHighlights`.

It is tricky to remove the highlight without unfocusing it in the UI, but I did sometimes encounter highlights getting "stuck focused" after scrolling up a few pages from where I had made them and then clicking the card in the UI. I haven't figured out exactly what is happening yet, but it is possible there is an issue where annotations are getting unnecessarily re-anchored in PDFs in some situation. This would have gone unnoticed before.

Reply:
This should be fixed now; highlights are explicitly un-focused before removal.